rocketchat/templates/chat-deployment.yaml

```diff
@@ -143,6 +143,42 @@ spec:
 {{- with .Values.extraEnv }}
 	{{- include "common.tplvalues.render" (dict "value" . "context" $) | nindent 8 }}
 {{- end }}
+        {{- if .Values.federation.enabled }}
+        - name: OVERWRITE_SETTING_Federation_Matrix_homeserver_domain
+          value: {{ .Values.host }}
+        - name: OVERWRITE_SETTING_Federation_Matrix_enabled
+          value: "true"
+        - name: OVERWRITE_SETTING_Federation_Matrix_serve_well_known
+          {{- if not .Values.ingress.federation.serveWellKnown }}
+          value: "true"
+          {{- else }}
+          value: "false"
```

Comment:
This isn't implemented yet, we will be able to avoid litghttpd and other ingress configs this way. 

pr isn't merged yet https://github.com/RocketChat/Rocket.Chat/pull/31965

Idea was to make this default for chart installs, but we need this now and can make it opt in.